pkg/sysinfo/sysinfo_linux.go

```diff
@@ -5,22 +5,62 @@ import (
 	"os"
 	"path"
 	"strings"
+	"sync"
 
-	cdcgroups "github.com/containerd/cgroups"
-	cdseccomp "github.com/containerd/containerd/pkg/seccomp"
-	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/containerd/cgroups"
+	"github.com/containerd/containerd/pkg/seccomp"
+	"github.com/moby/sys/mountinfo"
 	"github.com/sirupsen/logrus"
 )
 
-func findCgroupMountpoints() (map[string]string, error) {
-	cgMounts, err := cgroups.GetCgroupMounts(false)
+var (
+	readMountinfoOnce sync.Once
+	readMountinfoErr  error
+	cgroupMountinfo   []*mountinfo.Info
+)
+
+// readCgroupMountinfo returns a list of cgroup v1 mounts (i.e. the ones
+// with fstype of "cgroup") for the current running process.
+//
+// The results are cached (to avoid re-reading mountinfo which is relatively
+// expensive), so it is assumed that cgroup mounts are not being changed.
+func readCgroupMountinfo() ([]*mountinfo.Info, error) {
+	readMountinfoOnce.Do(func() {
+		cgroupMountinfo, readMountinfoErr = mountinfo.GetMounts(
+			mountinfo.FSTypeFilter("cgroup"),
+		)
+	})
+
+	return cgroupMountinfo, readMountinfoErr
+}
+
+func findCgroupV1Mountpoints() (map[string]string, error) {
+	mounts, err := readCgroupMountinfo()
+	if err != nil {
+		return nil, err
+	}
+
+	allSubsystems, err := cgroups.ParseCgroupFile("/proc/self/cgroup")
 	if err != nil {
 		return nil, fmt.Errorf("Failed to parse cgroup information: %v", err)
```

Comment:
(not for this PR) we should fix this error to not start with a capital, and use pkg/errors for this (not sure if we should `wrap` (as that may return a `ENOENT`) or keep formatting as-is.